### ISSUE TITLE
Document new autoscript behavior more fully

### ISSUE THREAD

**Issue by morsssss** [ampproject/amp-toolbox#589]
Can you explain more fully in the readme.md what the automatic script inclusion mechanism is, why it's useful, and if there are cases where it might fail?

Also can you document it on [NPM](https://www.npmjs.com/package/@ampproject/toolbox-optimizer)?

/cc @garanj . If amp-toolbox becomes the preferred tool here, we should mention that and link to it on [the npm page for amphtml-autoscript](https://www.npmjs.com/package/amphtml-autoscript).

**Comment by garanj** [comment]
Yes, please let me know. I stand ready to change the page for `amphtml-autoscript` and can deprecate as necessary.

I think it is better that there is one preferred tool than confusing the picture with two options. So let me know when to light the fuse...
